src/System.Private.DataContractSerialization/src/System/Xml/XmlStreamNodeWriter.cs

```diff
@@ -125,18 +149,38 @@ private void EnsureByte()
             }
         }
 
+        private async Task EnsureByteAsync()
+        {
+            if (_offset >= bufferLength)
+            {
+                await FlushBufferAsync();
+            }
+        }
+
         protected void WriteByte(byte b)
         {
             EnsureByte();
             _buffer[_offset++] = b;
         }
 
+        protected async Task WriteByteAsync(byte b)
+        {
+            await EnsureByteAsync();
+            _buffer[_offset++] = b;
+        }
```

Comment:
This would be more efficient as something like:
```C#
protected Task WriteByteAsync(byte b)
{
    if (_offset >= bufferLength)
    {
        return FlushBufferAndWriteByteAsync(b);
    }
    else
    {
        _buffer[_offset++] = b;
        return Task.CompletedTask;
    }
}

private async Task FlushBufferAndWriteByteAsync(byte b)
{
    await EnsureByteAsync().ConfigureAwait(false);
    _buffer[_offset++] = b;
}
```